### PR TITLE
Removed Decoder Bias/Mask

### PR DESCRIPTION
**Pull request by rwitten**
Additionally I had to tune the convergence test due to model configuration changes.

### CODE REVIEW

File: MaxText/layers/embeddings.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def __call__(
       a jax.Array of shape [B, S, N, H] which includes the inputs together with
       the rotary position embedding incorporated in it.
     """
+    assert position is not None
```
Comment (hanzlfs) [inline MaxText/layers/embeddings.py:147]: Or raise ValueError?

**Review by hanzlfs — APPROVED** [review]

File: MaxText/tests/llama_test.py [hunk 2]
```diff
@@ -93,12 +93,15 @@ def test_llama_rope(self):
...
     )
 
+    seq_length = x_q.shape[1]
+    position = jnp.arange(seq_length, dtype=jnp.float32)[jnp.newaxis, :]
+
     # Calculate RoPE embeddings from MaxText implementation
-    query_proj = embeddings.LLaMARotaryEmbedding(embedding_dims = dim_per_head)(x_q)
-    key_proj = embeddings.LLaMARotaryEmbedding(embedding_dims = dim_per_head)(x_k)
+    query_proj = embeddings.LLaMARotaryEmbedding(embedding_dims = dim_per_head)(x_q, position = position)
+    key_proj = embeddings.LLaMARotaryEmbedding(embedding_dims = dim_per_head)(x_k, position = position)
```
Comment (patemotter) [inline MaxText/tests/llama_test.py:104]: Nit: remove spaces around ` = `

File: end_to_end/test_convergence_1b_params.sh [hunk 3]
```diff
@@ -26,7 +26,7 @@ done
 # Train
 export LIBTPU_INIT_ARGS="--xla_tpu_enable_data_parallel_all_reduce_opt=true --xla_tpu_data_parallel_opt_different_sized_ops=true --xla_tpu_enable_async_collective_fusion=true --xla_tpu_enable_async_collective_fusion_fuse_all_gather=true --xla_tpu_enable_async_collective_fusion_multiple_steps=true --xla_tpu_overlap_compute_collective_tc=true --xla_enable_async_all_gather=true"
 python3 MaxText/train.py MaxText/configs/base.yml run_name=$RUN_NAME\
-    steps=3400 per_device_batch_size=12.0 learning_rate=1e-3 enable_checkpointing=false\
+    steps=20400 per_device_batch_size=2.0 learning_rate=1e-4 enable_checkpointing=false\
```
Comment (patemotter) [inline end_to_end/test_convergence_1b_params.sh:29]: Is this change to the default training args needed and/or does it belong as part of this commit?

**Review by patemotter — COMMENTED** [review]
Overall it LGTM. I added a few minor comments and questions.

File: MaxText/layers/attentions.py [hunk 4]
```diff
@@ -49,55 +49,15 @@
 LENGTH = common_types.LENGTH
 HEAD = common_types.HEAD
 D_KV = common_types.D_KV
+DEFAULT_MASK_VALUE = -0.7 * float(jnp.finfo(jnp.dtype("float32")).max)
```
[the remark this answers is not in the document]
Comment (rwitten) [reply MaxText/layers/attentions.py:52]: Based on pallas flash attention kernel. Comment added.

File: MaxText/layers/attentions.py [hunk 5]
```diff
@@ -298,6 +279,8 @@ def apply_attention(
     if not deterministic and self.dropout_rate > 0.:
       attn_weights = self.attention_dropout(attn_weights, dropout_rng)
 
+
+
```
[the remark this answers is not in the document]
Comment (rwitten) [reply MaxText/layers/attentions.py:283]: ack

File: MaxText/layers/attentions.py [hunk 6]
```diff
@@ -469,8 +426,6 @@ def __call__(self,
     Args:
```
[the remark this answers is not in the document]
Comment (rwitten) [reply MaxText/layers/attentions.py:426]: ack

File: MaxText/layers/embeddings.py [hunk 7]
```diff
@@ -128,7 +128,7 @@ def setup(self) -> None:
   def __call__(
       self,  # pytype: disable=signature-mismatch  # overriding-parameter-count-checks
       inputs: jax.Array,
-      position: Optional[jax.Array] = None,
+      position: jax.Array,
```
[the remark this answers is not in the document]
Comment (rwitten) [reply MaxText/layers/embeddings.py:131]: ack

File: end_to_end/test_convergence_1b_params.sh (hunk 3 above)
Comment (rwitten) [reply end_to_end/test_convergence_1b_params.sh:29]: This change is driven by other changes to the model. And unfortunately we aren't yet running this test on a continuous basis
